### ISSUE TITLE
Milestone should be assigned to PRs when they are merged, not when they are approved

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#6516]
In https://github.com/woocommerce/automations/pull/41 we added an automation to automatically assign the milestone to a PR once it's approved. However, it would be better to assign it once the PR is merged. The main reason is that PRs might be approved, but they might not be ready for release yet (ie: there are conflicts, the PR needs to be released with a pairing PR, the PR was pre-approved but there are still changes needed, etc.).

This issue is about updating the automation so PRs are assigned to the next milestone when and only when they are merged into `trunk`.

**Comment by alexflorisca** [comment]
I think either team could work on this one but I see you've assigned it to Kirigami Backlog so will let you guys handle it :) If you need a hand with it let us know

**Comment by mikejolley** [comment]
Closing due to pca54o-6Dz-p2
